### DISCUSSION TITLE
Mogrify different behaviour between different versions

### DISCUSSION THREAD

**Discussion by raghavsvasan** [ImageMagick/ImageMagick#1899] [General]
Hi all
I am a beginner with Imagemagick, but I have been using mogrify command to convert a bunch of pdf files to png on my mac with Version: ImageMagick 6.9.9-26 Q16 x86_64 2018-02-22 

I have been trying to replicate this in a linux server with Version: ImageMagick 6.5.4-7 2012-04-10 Q16 OpenMP 

I am getting a significant difference in the resultant pngs produced. 
For eg, mac 6.9 version produced pngs with color profile of sRGB whereas the linux version did not. Also mac version had a 8 bit alpha channel and the linux version did not. Even if I tried to add alpha channel it produced only a 1 bit alpha channel

I was trying different options like setting depth to 8 and alpha to activate but I could not match the exact image format produced in 6.9 version mogrify.

Request your suggestions on how to replicate the exact format from my mac without upgrading the linux imagemagick version.

My command in linux is
 mogrify -format png -quality 100 -depth 8 -profile sRGB.icc -alpha activate A*.pdf - could not match the image format.

My command in mac is
mogrify -format png -quality 100 A*.pdf

My required image format is as below (produced in mac 6.9 version of imagemagick) is 

Image: A53026_kainrl-04Mar.png
  Format: PNG (Portable Network Graphics)
  Mime type: image/png
  Class: DirectClass
  Geometry: 612x792+0+0
  Resolution: 72x72
  Print size: 8.5x11
  Units: Undefined
  Type: TrueColorAlpha
  Endianess: Undefined
  Colorspace: sRGB
  Depth: 8-bit
  Channel depth:
    red: 8-bit
    green: 8-bit
    blue: 8-bit
    alpha: 8-bit
  Channel statistics:
    Pixels: 484704
    Red:
      min: 0  (0)
      max: 255 (1)
      mean: 223.128 (0.875011)
      standard deviation: 83.0939 (0.325858)
      kurtosis: 3.18702
      skewness: -2.26474
      entropy: 0.0923133
    Green:
      min: 0  (0)
      max: 255 (1)
      mean: 221.386 (0.868182)
      standard deviation: 85.0745 (0.333626)
      kurtosis: 2.75677
      skewness: -2.16823
      entropy: 0.099373
    Blue:
      min: 0  (0)
      max: 255 (1)
      mean: 223.836 (0.877788)
      standard deviation: 81.7084 (0.320425)
      kurtosis: 3.44379
      skewness: -2.31291
      entropy: 0.0942768
    Alpha:
      min: 255  (1)
      max: 0 (0)
      mean: 20.8204 (0.0816487)
      standard deviation: 58.7847 (0.230528)
      kurtosis: 7.23336
      skewness: -2.8895
      entropy: 0.164629
  Image statistics:
    Overall:
      min: 255  (1)
      max: 0 (0)
      mean: 29.3676 (0.115167)
      standard deviation: 78.062 (0.306126)
      kurtosis: 3.9569
      skewness: -2.40391
      entropy: 0.112648
  Alpha: srgba(255,255,255,0)   #FFFFFF00
  Rendering intent: Perceptual
  Gamma: 0.454545
  Chromaticity:
    red primary: (0.64,0.33)
    green primary: (0.3,0.6)
    blue primary: (0.15,0.06)
    white point: (0.3127,0.329)
  Background color: white
  Border color: srgba(223,223,223,1)
  Matte color: grey74
  Transparent color: none
  Interlace: None
  Intensity: Undefined
  Compose: Over
  Page geometry: 612x792+0+0
  Dispose: Undefined
  Iterations: 0
  Compression: Zip
  Orientation: Undefined
  Properties:
    date:create: 2020-04-14T10:32:19+12:00
    date:modify: 2020-04-09T20:12:16+12:00
    icc:copyright: Copyright Artifex Software 2011
    icc:description: Artifex Software sRGB ICC Profile
    icc:manufacturer: Artifex Software sRGB ICC Profile
    icc:model: Artifex Software sRGB ICC Profile
    pdf:HiResBoundingBox: 612x792+0+0
    pdf:Version: PDF-1.4 
    png:bKGD: chunk was found (see Background color, above)
    png:cHRM: chunk was found (see Chromaticity, above)
    png:iCCP: chunk was found
    png:IHDR.bit-depth-orig: 8
    png:IHDR.bit_depth: 8
    png:IHDR.color-type-orig: 6
    png:IHDR.color_type: 6 (RGBA)
    png:IHDR.interlace_method: 0 (Not interlaced)
    png:IHDR.width,height: 612, 792
    png:pHYs: x_res=72, y_res=72, units=0
    png:text: 4 tEXt/zTXt/iTXt chunks were found
    signature: 693c2a01b98993d111ca6f67d0242b186a97162542ee0145c7ef55b0f3cf57f6
  Profiles:
    Profile-icc: 2576 bytes
  Artifacts:
    filename: A53026_kainrl-04Mar.png
    verbose: true
  Tainted: False
  Filesize: 91706B
  Number pixels: 484704
  Pixels per second: 48.4704MB
  User time: 0.020u
  Elapsed time: 0:01.009
  Version: ImageMagick 6.9.9-26 Q16 x86_64 2018-02-22 http://www.imagemagick.org

**Comment by dlemstra** [comment]
Version `6.5.4-7` is ancient and I would advise you to upgrade the version on your Linux server. And you should also check if you are using the same version of Ghostscript on both systems?

**Reply by raghavsvasan** [reply]
Thanks for that. Agreed. Unfortunately Im unable to do the upgrade straightaway. Is there any other way to solve this difference ?

**Comment by raghavsvasan** [comment]
Forgot to mention in my first post that even the statistics for RGB channels are different between these versions. Is this because of different compression levels? Mac version has lower RGB mean values compared to its linux version image.

**Reply by dlemstra** [reply]
A lot of changes have been made to the software over the years. And the only way to resolve this would be upgrading to a more recent version of ImageMagick.
